x/subscription/client/cli/tx.go

```diff
@@ -0,0 +1,258 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+
+	"github.com/crypto-com/chain-main/x/subscription/types"
+)
+
+// Proposal flags
+const (
+	FlagTitle          = "title"
+	FlagDescription    = "description"
+	FlagPrice          = "price"
+	flagDuration       = "duration"
+	flagCronSpec       = "cron_spec"
+	flagPlanID         = "plan_id"
+	flagSubscriptionID = "subscription_id"
+	flagOwner          = "owner"
+	flagSubscriber     = "subscriber"
+)
+
+func NewTxCmd() *cobra.Command {
+	subscriptionTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Subscription transactions subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	subscriptionTxCmd.AddCommand(
+		NewCmdCreatePlan(),
+		NewCmdStopPlan(),
+		NewCmdCreateSubscription(),
+		NewCmdStopSubscription(),
+		NewCmdStopUserSubscription(),
+	)
+
+	return subscriptionTxCmd
+}
+
+// NewCmdCreatePlan implements submitting a proposal transaction command.
+func NewCmdCreatePlan() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-plan",
+		Short: "Create a subscription plan",
+		Long:  "",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			// Get voting address
+			owner := clientCtx.GetFromAddress()
+
+			flags := cmd.Flags()
+
+			title, err := flags.GetString(FlagTitle)
+			if err != nil {
+				return err
+			}
+			description, err := flags.GetString(FlagDescription)
+			if err != nil {
+				return err
+			}
+			strPrice, err := flags.GetString(FlagPrice)
+			if err != nil {
+				return err
+			}
+			price, err := sdk.ParseCoinsNormalized(strPrice)
+			if err != nil {
+				return err
+			}
+			duration, err := flags.GetUint32(flagDuration)
+			if err != nil {
+				return err
+			}
+			strSpec, err := flags.GetString(flagCronSpec)
+			if err != nil {
+				return err
+			}
+			spec, err := types.ParseCronSpec(strSpec)
+			if err != nil {
+				return err
+			}
+			msg := types.MsgCreatePlan{
+				Owner:                owner.String(),
+				Title:                title,
+				Description:          description,
+				Price:                price,
+				SubscriptionDuration: duration,
+				CronSpec:             spec,
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return fmt.Errorf("message validation failed: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+
+	cmd.Flags().String(FlagTitle, "", "The plan title")
+	cmd.Flags().String(FlagDescription, "", "The plan description")
+	cmd.Flags().String(FlagPrice, "", "The plan price")
+	cmd.Flags().Uint32(flagDuration, 365*24*3600, "The plan subscription duration")
```

Comment:
Unit of duration?

Reply:
it's in seconds, I renamed it to make it clear.